skyrim-platform/src/platform_lib/HttpClientApi.cpp

```diff
@@ -72,18 +81,18 @@ JsValue HttpClientApi::Get(const JsFunctionArguments& args)
 
 JsValue HttpClientApi::Post(const JsFunctionArguments& args)
 {
-  JsValue path = args[1], body = args[2], contentType = args[3],
-          headers = args[4], host = args[0].GetProperty("host");
+  JsValue path = args[1], options = args[2],
+          host = args[0].GetProperty("host");
 
   auto resolverFn = JsValue::Function([=](const JsFunctionArguments& args) {
     auto resolve = std::make_shared<JsValue>(args[1]);
     auto pathStr = static_cast<std::string>(path);
-    auto bodyStr = static_cast<std::string>(body);
     auto hostStr = static_cast<std::string>(host);
-    auto contentTypeStr = static_cast<std::string>(contentType);
+    auto bodyStr = static_cast<std::string>(options.GetProperty("body"));
```

Comment:
Does it throw or reject promise if `options = undefined`?

Reply:
`GetProperty` throws immediately.

Reply:
C++ exception? I think it should result in promise rejection. If that's the case, then perfect that it works out of the box (though error message would be not quite human-friendly, I guess)

Reply:
It isn't super-readable by design. Since the main language of Skyrim Platform scripting is TypeScript, we expect to never encounter such type errors. Code with incorrect types won't even compile.